packages/node/src/login/oidc/oidcHandlers/ClientCredentialsOidcHandler.spec.ts

```diff
@@ -0,0 +1,359 @@
+      Promise.resolve("https://my.webid/")
+    ),
+  };
+});
+
+const mockJwk = (): JWK => {
+  return {
+    kty: "EC",
+    kid: "oOArcXxcwvsaG21jAx_D5CHr4BgVCzCEtlfmNFQtU0s",
+    alg: "ES256",
+    crv: "P-256",
+    x: "0dGe_s-urLhD3mpqYqmSXrqUZApVV5ZNxMJXg7Vp-2A",
+    y: "-oMe9gGkpfIrnJ0aiSUHMdjqYVm5ZrGCeQmRKoIIfj8",
+    d: "yR1bCsR7m4hjFCvWo8Jw3OfNR4aiYDAFbBD9nkudJKM",
+  };
+};
+
+const mockIdToken = (): string =>
+  "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJodHRwczovL215LndlYmlkIiwiaXNzIjoiaHR0cHM6Ly9teS5pZHAvIiwiYXVkIjoiaHR0cHM6Ly9yZXNvdXJjZS5leGFtcGxlLm9yZyIsImV4cCI6MTY2MjI2NjIxNiwiaWF0IjoxNDYyMjY2MjE2fQ.IwumuwJtQw5kUBMMHAaDPJBppfBpRHbiXZw_HlKe6GNVUWUlyQRYV7W7r9OQtHmMsi6GVwOckelA3ErmhrTGVw";
+
+type AccessJwt = {
+  sub: string;
+  iss: string;
+  aud: string;
+  nbf: number;
+  exp: number;
+  cnf: {
+    jkt: string;
+  };
+};
+
+const mockWebId = (): string => "https://my.webid/";
+
+const mockKeyBoundToken = (): AccessJwt => {
+  return {
+    sub: mockWebId(),
+    iss: mockDefaultIssuerConfig().issuer.toString(),
+    aud: "https://resource.example.org",
+    nbf: 1562262611,
+    exp: 1562266216,
+    cnf: {
+      jkt: mockJwk().kid as string,
+    },
+  };
+};
+
+const mockIdTokenPayload = (): IdTokenClaims => {
+  return {
+    sub: "https://my.webid/",
+    iss: "https://my.idp/",
+    aud: "https://resource.example.org",
+    exp: 1662266216,
+    iat: 1462266216,
+  };
+};
+
+const mockDpopTokens = (): TokenSet => {
+  return {
+    access_token: JSON.stringify(mockKeyBoundToken()),
+    id_token: mockIdToken(),
+    token_type: "DPoP",
+    expired: () => false,
+    claims: mockIdTokenPayload,
+  };
+};
+
+const mockBearerTokens = (): TokenSet => {
+  return {
+    access_token: "some token",
+    id_token: mockIdToken(),
+    token_type: "Bearer",
+    expired: () => false,
+    claims: mockIdTokenPayload,
+  };
+};
+
+const setupOidcClientMock = (tokenSet: TokenSet) => {
+  const { Issuer } = jest.requireMock("openid-client") as any;
+  function clientConstructor() {
+    // this is untyped, which makes TS complain
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.grant = jest.fn().mockResolvedValueOnce(tokenSet);
+  }
+  const mockedIssuer = {
+    metadata: mockDefaultIssuerConfig(),
+    Client: clientConstructor,
+  };
+  Issuer.mockReturnValueOnce(mockedIssuer);
+};
+
+describe("ClientCredentialsOidcHandler", () => {
+  describe("canHandle", () => {
+    it("cannot handle if the client ID is missing", async () => {
+      const clientCredentialsOidcHandler = new ClientCredentialsOidcHandler(
+        mockDefaultTokenRefresher(),
+        mockStorageUtility({})
+      );
+      await expect(
+        clientCredentialsOidcHandler.canHandle({
+          ...standardOidcOptions,
+          client: {
+            clientId: undefined as unknown as string,
+            clientType: "static",
+          },
+        })
+      ).resolves.toEqual(false);
```

Comment:
Also for the other tests; `toEqual` also checks object properties.

```suggestion
      ).resolves.toBe(false);
```